src/data/model/UserOrganizationMembershipRole.ts

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2023 PixieBrix, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { type components } from "@/types/swagger";
+import { UserRole } from "@/types/contract";
+
+export type UserOrganizationMembershipRole =
+  | "member"
+  | "admin"
+  | "developer"
+  | "restricted"
+  | "manager";
+
+export function transformUserOrganizationMembershipRoleResponse(
+  response: components["schemas"]["Me"]["organization_memberships"][number]["role"],
+): UserOrganizationMembershipRole {
+  switch (response) {
+    case 1: {
+      return "member";
+    }
+
+    case 2: {
+      return "admin";
+    }
+
+    case 3: {
+      return "developer";
+    }
+
+    case 4: {
+      return "restricted";
+    }
+
+    case 5: {
+      return "manager";
+    }
+
+    default: {
+      throw new Error(
+        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions -- future-proofing
+        `Invalid user organization membership role: ${response}`,
```

Comment:
This will throw an error during development if we ever add an additional role to the enum but don't add it to the switch:
```suggestion
        const role: never = response;
        `Invalid user organization membership role: ${response}`,
```